packages/table/test/loadingOptionsTests.tsx

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as React from "react";
+
+import { Cell,
+    Column,
+    ColumnHeaderCell,
+    ColumnLoadingOption,
+    RowHeaderCell,
+    RowLoadingOption,
+    Table,
+    TableLoadingOption,
+} from "../src";
+import { CellType, expectCellLoading } from "./cellTestUtils";
+import { ReactHarness } from "./harness";
+
+describe("Loading Options", () => {
+    const harness = new ReactHarness();
+    const allTableLoadingOptions = generatePowerSet([
+        TableLoadingOption.CELLS,
+        TableLoadingOption.COLUMN_HEADERS,
+        TableLoadingOption.ROW_HEADERS,
+    ]);
+    const allColumnLoadingOptions = generatePowerSet([
+        ColumnLoadingOption.CELLS,
+        ColumnLoadingOption.HEADER,
+    ]);
+    const allRowLoadingOptions = generatePowerSet([
+        RowLoadingOption.CELLS,
+        RowLoadingOption.HEADER,
+    ]);
+
+    afterEach(() => {
+        harness.unmount();
+    });
+
+    after(() => {
+        harness.destroy();
+    });
+
+    /*
+     * Cell loading overrides column loading which in turn overrides table loading. If loading
+     * options are omitted, then the loading options of the parent component are used. Because there
+     * is no `Row` component, row header cell loading overrides table loading. Below is an
+     * exhaustive set of tests of all possible combinations of loading options.
+     */
+    allTableLoadingOptions.forEach((tableLoadingOptions: TableLoadingOption[]) => {
+        allColumnLoadingOptions.forEach((columnLoadingOptions: ColumnLoadingOption[]) => {
+            allRowLoadingOptions.forEach((rowLoadingOptions: RowLoadingOption[]) => {
```

Comment:
shouldn't need types on the params here. can they not be inferred from `forEach`?